ctapipe/core/traits.py

```diff
@@ -270,38 +297,40 @@ def __init__(self, dtype=float, default_value=None, **kwargs):
 
     def validate(self, obj, value):
         # Support a single value for all (convert into a default value)
+        print(f"VALIDATE: obj={obj}, value={value} ({type(value)}")
         if isinstance(value, self._dtype):
             value = [("type", "*", value)]
 
         # Check each value of list
-        normalized_value = []
-        for pattern in value:
-            # now check for the standard 3-tuple of (command, argument, value)
-            if len(pattern) != 3:
-                raise TraitError(
-                    "pattern should be a tuple of (command, argument, value)"
-                )
-            command, arg, val = pattern
-            if not isinstance(val, self._dtype):
-                raise TraitError(f"Value should be a {self._dtype}")
-            if not isinstance(command, str):
-                raise TraitError("command must be a string")
-            if command not in ["type", "id"]:
-                raise TraitError("command must be one of: '*', 'type', 'id'")
-            if command == "type":
-                if not isinstance(arg, str):
-                    raise TraitError("'type' argument should be a string")
-            if command == "id":
-                arg = int(arg)
-
-            val = self._dtype(val)
-            normalized_value.append((command, arg, val))
-
-        # Convert to TelescopeParameterLookup
-        normalized_value = TelescopeParameterLookup(normalized_value)
-
-        # Validate with super method
-        super().validate(obj, normalized_value)
+        normalized_value = TelescopePatternList(None)
+        if isinstance(value, self._dtype):
+            value = [("type", "*", value)]
+        if isinstance(value, UserList) or isinstance(value, list):
+            for pattern in value:
+                # now check for the standard 3-tuple of (command, argument, value)
+                if len(pattern) != 3:
+                    raise TraitError(
+                        "pattern should be a tuple of (command, argument, value)"
+                    )
+                command, arg, val = pattern
+                if not isinstance(val, self._dtype):
+                    raise TraitError(f"Value should be a {self._dtype}")
+                if not isinstance(command, str):
+                    raise TraitError("command must be a string")
+                if command not in ["type", "id"]:
+                    raise TraitError("command must be one of: '*', 'type', 'id'")
+                if command == "type":
+                    if not isinstance(arg, str):
+                        raise TraitError("'type' argument should be a string")
+                if command == "id":
+                    arg = int(arg)
```

Comment:
All other raise `TraitError`, but this will raise `ValueError` if the id arg is not convertible to integer